haystack/modeling/model/feature_extraction.py

```diff
@@ -31,45 +39,112 @@
-    Right now it always returns an instance of `AutoTokenizer`.
-
-    :param pretrained_model_name_or_path:  The path of the saved pretrained model or its name (e.g. `bert-base-uncased`)
-    :param revision: The version of model to use from the HuggingFace model hub. Can be tag name, branch name, or commit hash.
-    :param use_fast: Indicate if Haystack should try to load the fast version of the tokenizer (True) or use the Python one (False). Defaults to True.
-    :param use_auth_token: The API token used to download private models from Huggingface.
-                           If this parameter is set to `True`, then the token generated when running
-                           `transformers-cli login` (stored in ~/.huggingface) will be used.
-                           Additional information can be found here
-                           https://huggingface.co/transformers/main_classes/model.html#transformers.PreTrainedModel.from_pretrained
-    :param kwargs: other kwargs to pass on to `PretrainedTokenizer.from_pretrained()`
-    :return: AutoTokenizer instance
-    """
-    model_name_or_path = str(pretrained_model_name_or_path)
+FEATURE_EXTRACTORS = {
+    **{key: AutoTokenizer for key in TOKENIZER_MAPPING_NAMES.keys()},
+    **{key: AutoFeatureExtractor for key in FEATURE_EXTRACTOR_MAPPING_NAMES.keys()},
+    # "clip": CLIPProcessor,
+}
+
 
-    if "mlm" in model_name_or_path.lower():
-        logging.error("MLM part of codebert is currently not supported in Haystack. Proceed at your own risk.")
+DEFAULT_EXTRACTION_PARAMS = {
+    AutoTokenizer: {
+        "max_length": 256,
+        "add_special_tokens": True,
+        "truncation": True,
+        "truncation_strategy": "longest_first",
+        "padding": "max_length",
+        "return_token_type_ids": True,
+    },
+    AutoFeatureExtractor: {"return_tensors": "pt"},
+}
 
-    params = {}
-    if any(tokenizer_type in model_name_or_path for tokenizer_type in ["albert", "xlnet"]):
-        params["keep_accents"] = True
 
-    return AutoTokenizer.from_pretrained(
-        pretrained_model_name_or_path=model_name_or_path,
-        revision=revision,
-        use_fast=use_fast,
-        use_auth_token=use_auth_token,
-        **params,
+def _safe_tensor_conversion(features: Dict[str, Any]):
+    """
+    Converts all features into tensors if all input values are 2D integer vectors.
+    """
+    for tensor_name, tensors in features.items():
+        sample = tensors[0][0]
+
+        # Check that the cast to long is safe
+        if not np.issubdtype(type(sample), np.integer):
+            raise ModelingError(
+                f"Feature '{tensor_name}' (sample value: {sample}, type: {type(sample)})"
+                " can't be converted safely into a torch.long type."
+            )
+        # Cast all data to long
+        tensors = torch.as_tensor(np.array(tensors), dtype=torch.long)
+        features[tensor_name] = tensors
+    return features
+
+
+class FeatureExtractor:
```

Comment:
@ZanSara, I really like this generalization in the form of FeatureExtractor. Great work